util/test/sub_test

```diff
@@ -579,6 +579,45 @@ def runSkipIf(skipifName):
 
     return stdout
 
+# Translate some known failures into more easily understood forms
+def translateOutput(output):
+    xlates = (('slurmstepd: Munge decode failed: Expired credential',
+               'Jira 18 -- Expired slurm credential for'),
+              ('output file from job .* does not exist',
+               'Jira 17 -- Missing output file for'),
+              ('aprun: Unexpected close of the apsys control connection',
+               'Jira 193 -- Unexpected close of apsys for'),
+              ('Transient MPP reservation error on create',
+               'Jira 203 -- Transient MPP reservation error for'),
+              ('qsub: cannot connect to server sdb',
+               'Jira 283 -- Sporadic: qstat failed to connect to server sdb'),
+              ('Failed to recv data from background qsub',
+               'Jira 260 -- Failed to recv data from background qsub for'),
+              (r'\d+ Killed /var/spool/PBS/mom_priv/jobs',
+               'Jira 324 -- PBS job killed for'),
+              ('Fatal MPP reservation error on confirm',
+               'Jira 329 -- Fatal MPP reservation error for'),
+              ('Fatal MPP reservation error on create',
+               'Jira 329 -- Fatal MPP reservation error for'),
+              ('Text file busy',
+               'Jira 333 -- Text file busy for'))
+
+    known_error = ''
+    was_timeout = False
+
+    for x in xlates:
+        if re.search(x[0], output, re.IGNORECASE) != None:
+            known_error = x[1]
+            break
+    else:
```

Comment:
What `if` is this matching up against?

Reply:
It's matching the `for`.  If you put an `else` on a loop-stmt that `break`s, it will run when the loop-stmt goes to completion, i.e. if the `break` doesn't happen.

Reply:
Oh, nevermind. I didn't realize for/else was a thing. Cool